src/test/java/org/junit/rules/TemporaryFolderUsageTest.java

```diff
@@ -113,22 +124,46 @@ public void newFolderWithPathContainingForwardSlashCreatesDirectories()
     }
     
     @Test
-    public void newFolderWithGivenPathThrowsIllegalArgumentExceptionIfPathExists() throws IOException {
+    public void newFolderWithGivenPathThrowsIllegalArgumentExceptionIfFolderExists() throws IOException {
         tempFolder.create();
         tempFolder.newFolder("level1", "level2", "level3");
 
         thrown.expect(IOException.class);
-        thrown.expectMessage("a folder with the name 'level3' already exists");
+        String path = "level1" + File.separator + "level2" + File.separator + "level3";
+        thrown.expectMessage("a folder with the path '" + path + "' already exists");
         tempFolder.newFolder("level1", "level2", "level3");
     }
 
     @Test
-    public void newFolderWithGivenPathThrowsIOExceptionIfFolderNamesConsistOfMultiplePathComponents()
+    public void newFolderWithPathsContainingForwardSlashCreatesFullPath()
             throws IOException {
         tempFolder.create();
-        thrown.expect(IOException.class);
-        thrown.expectMessage("name cannot consist of multiple path components");
         tempFolder.newFolder("temp1", "temp2", "temp3/temp4");
+
+        File directory = new File(tempFolder.getRoot(), "temp1");
+        assertFileIsDirectory(directory);
+        directory = new File(directory, "temp2");
+        assertFileIsDirectory(directory);
+        directory = new File(directory, "temp3");
+        assertFileIsDirectory(directory);
+        directory = new File(directory, "temp4");
+        assertFileIsDirectory(directory);
+    }
+
+    @Test
+    public void newFolderWithPathsContainingFileSeparatorCreatesFullPath()
+            throws IOException {
+        tempFolder.create();
+        tempFolder.newFolder("temp1", "temp2", "temp3" + File.separator + "temp4");
+
+        File directory = new File(tempFolder.getRoot(), "temp1");
+        assertFileIsDirectory(directory);
+        directory = new File(directory, "temp2");
+        assertFileIsDirectory(directory);
+        directory = new File(directory, "temp3");
+        assertFileIsDirectory(directory);
+        directory = new File(directory, "temp4");
+        assertFileIsDirectory(directory);
```

Comment:
Wouldn't it be sufficient have a single assertion: `assertFileIsDirectory(new File(tempFolder.getRoot(), "temp1/temp2/temp3/temp4")`?